lib/grpnice.gi

```diff
@@ -16,11 +16,22 @@
 ##
 #M  SetNiceMonomorphism(<G>,<hom>)
 ##
-##  This setter method is special because we want to tell every nice
-##  monomorphism that it is one.
+##  We install a special setter method because we have to make sure that only
+##  injective maps get stored as nice monomorphisms.
+##  More precisely, we the stored map must know that it is injective,
+##  otherwise computing its kernel may run into an infinite recursion.
+##  (The maps computed by 'NiceMonomorphism' have the 'IsInjective' flag,
+##  the test affects only those maps that shall be set by hand as
+##  nice monomorphisms.)
```

Comment:
Strictly speaking this is just the case for all the *current* such methods. Anyone adding more such methods in the future should also be aware of it, I think? So perhaps the sentence here should either be changed to acknowledge that -- or just remove it? I don't think it is all that useful: we want to enforce an invariant, period. That some code is (supposedly) already correctly enforcing the invariant is immaterial, isn't it?